rfcs/0015-consistent-environment-naming.md

```diff
@@ -0,0 +1,84 @@
+- Feature Name: consistent-environment-naming
+- Start Date: 2022-09-14
+- RFC PR: [mbta/technology-docs#0015](https://github.com/mbta/technology-docs/pull/15)
+- Asana task: [Dev environment naming / usage conventions](https://app.asana.com/0/1200506724882024/1202675232761300/f)
+- Status: Proposed
+
+# Summary
+[summary]: #summary
+
+Non-production environments of CTD applications have clear, descriptive, and consistent naming.
+
+# Motivation
+[motivation]: #motivation
+
+Non-production environments for our applications are generally named with some variation on `dev`. This doesn't capture the distinctions between different non-production uses of application environments.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Environments are named in `kebab-case`, following the convention `[name of application]-[modifier]`. Available modifiers are as follows, with a brief description of their use case:
+
+- `prod`: the production environment.
+- `staging`: used for final integration testing and monitoring for errors before deploying changes to production. Often, merging a PR into the `main` / `master` branch will be set up to automatically initiate a deploy to the application's `staging` environment. There should be one stage environment per application.
```

Comment:
That's a really good point. I wonder if there's another suffix we could use. Maybe `internal` or something, to indicate that it's not staging but it's still not quite "real" prod? @ianwestcott do you have any opinions on this? Are there any reasons that it would be inappropriate to just put `internal` right there in the domain name?